libcontainer/specconv/spec_linux.go

```diff
@@ -256,6 +267,8 @@ func CreateLibcontainerConfig(opts *CreateOpts) (*configs.Config, error) {
 	config.Cgroups = c
 	// set linux-specific config
 	if spec.Linux != nil {
+		initMaps()
```

Comment:
I like the improvement that's brought by this, but (as a follow-up) perhaps we should consider having utilities / accessor functions for these variables, so that the initialisation happens more 'organically' (reducing the cognitive overload of having to thing about _first_ calling `initMap()` whenever we need to use one of the variables.

In this case it could be a function to check if `foo` exists in `mountPropagationMapping`

Reply:
Do you mean something like
```
func getNamespaceMapping() map[specs.LinuxNamespaceType]configs.NamespaceType {
    initMaps()
    return namespaceMapping
}

// ... and so on for every other map we lazy init.
....

nsMap := getNamespaceMapping()
```

To me it looks like a bloat, since the `nsMap := getNamespaceMapping()` is just a way to sugar-coat `initMaps()`.

A runtime assertion would be nice. If only accessing a nil map would panic....

Reply:
I can't think of anything better than
```go
if namespaceMapping == nil {
   panic("uninitialied map") // we have a bug in the code
}
```

but it's still inferior to
```go
initMaps()
```

Reply:
Some kind of a linter to which we can add a rule (in the source code) like "X should be called before using Y" would also be nice to have.

Reply:
Ideally, a lazy way to initialize variables. IOW same as this PR does, but automatically (perhaps via an annotation).

Reply:
Someone worked on something like this -- alas it does not look good without generics. https://github.com/zetamatta/go-lazy

I guess, let's postpone it will Go 2.0 ))))